src/lib.rs

```diff
@@ -16,13 +16,13 @@ pub type Data<T> = <T as GraphObject>::Data;
 /// Abstract object in a graph, eg. node or edge.
 pub trait GraphObject {
     /// Identifier of all graph objects.
-    type Id: Clone;
+    type Id;
 
     /// Local graph object data.
     type Data;
 
     /// Return the edge id.
-    fn id(&self) -> Self::Id;
+    fn id(&self) -> &Self::Id;
```

Comment:
Same comment as above basically - I think this lowers ease of use, because you're now holding a reference to the container.